src/index.html

```diff
@@ -11,7 +11,184 @@
+          <label for="userAge">How old are You?</label>
+          <input 
+            type="number" 
+            name="personal"
+            id="userAge" 
+            value="12"
+            min="1"
+            max="100"
+          >
+        </div>
+        
+        <div class="fieldset__item">
+          <label for="dateOfBirth">Full date of birth:</label>
+          <input 
+            type="date" 
+            name="dateOfBirth"
+            id="dateOfBirth"
+          >
+        </div>
+
+        <div class="fieldset__item">
+          <label for="agreement">I accept the term of the agreement</label>
+          <input 
+            type="checkbox" 
+            name="agreement"
+            id="agreement" 
+            required
+          >
+        </div>
+        
+      </fieldset>
+
+      <fieldset class="fieldset">
+        <legend>Registration</legend>
+
+        <div class="fieldset__item">
+          <label for="userEmail">E-mail:</label>
+          <input 
+            type="email" 
+            name="userEmail"
+            id="userEmail"
+            placeholder="email@example.com"
+          >
+        </div>
+
+        <div class="fieldset__item">
+          <label for="userPassword">Password:</label>
+          <input 
+            type="password" 
+            name="userPassword"
+            id="userPassword"
+            minlength="8"
+            maxlength="16"
+          >
+        </div>
+      </fieldset>
+
+      <fieldset class="fieldset">
+        <legend>An interesting fact about you!</legend>
+
+        <div class="fieldset__item">
+          <label>Do you love cats? </label>
+
+          <input 
+          type="radio" 
+          name="cats"
+          id="confirmed"
+          value="Yes" 
+          >
+          <label for="confirmed">Yes</label>
+
+          <input 
+          type="radio" 
+          name="cats"
+          id="rejected"
+          value="no" 
+          >
+          <label for="rejected">No</label>
+        </div>
+
+        <div class="fieldset__item">
+          <label for="favouriteColor">What is your favorite color?</label>
+          <input 
+            type="color" 
+            name="favouriteColor"
+            id="favouriteColor"
+          >
+        </div>
+        
+        <div class="fieldset__item">
+          <label for="sleepingTime">What time do you go to bed?</label>
+          <input 
+            type="time" 
+            name="sleepingTime"
+            id="sleepingTime" 
+            step="2"
+          >
+        </div>
+        
+        <div class="fieldset__item">
+          <label for="favouriteCar">What are your favorite brand of cars?</label>
+
+          <select 
+            name="favouriteCar" 
+            id="favouriteCar" 
+            multiple
+          >
+            <option value="BMW">BMW</option>
+
+            <option value="Audi">Audi</option>
+            
+            <option value="Lada">Lada</option>
```

Comment:
since value attribute is passed on to the server - better use lower-case values to minimize any upper/lower-case-driven issues.
```suggestion
            <option value="bmw">BMW</option>
            <option value="audi">Audi</option>
            <option value="lada">Lada</option>
```